spec/controllers/programs_controller_spec.rb

```diff
@@ -9,54 +9,29 @@
     double(:panoptes_application_client)
   end
 
+  describe "GET methods" do
+    let!(:program) { create(:program) }
+    it 'GET index returns a list of programs' do
+      get :index, format: :json
+      expect(response.body).to eq(ActiveModelSerializers::SerializableResource.new([program]).to_json)
+    end
+
+    it 'GET show returns a single program' do
+      get :show, params: {id: program.id}, format: :json
+      expect(response.status).to eq(200)
+      expect(response.body).to eq(ActiveModelSerializers::SerializableResource.new(program).to_json)
+    end
+  end
+
   describe "POST create" do
     before { allow(controller).to receive(:panoptes_application_client).and_return(panoptes_application_client) }
 
     it 'returns a new program' do
       program = build(:program)
-      outcome = double(result: program, valid?: true)
       attributes = {slug: program.slug, name: program.name}
 
-      # Why is this spec busted?
-      # rebuilt = {}.tap do |param|
-      #   param[:attributes] = attributes
-      #   param[:current_user] = current_user
-      #   param[:client] = panoptes_application_client
-      # end
-      # action_params = ActionController::Parameters.new(rebuilt)
-
-      # expect(Programs::Create).to receive(:run)
-      #   .with(action_params)
-      #   .and_return(outcome)
-
-      post :create, params: {data: {attributes: attributes}}, as: :json
-      expect(parsed_response[:slug]).to eq(program.slug)
-    end
-  end
-
-  describe "PUT update" do
-    let(:new_name) { "A new and exciting name" }
-
-    it 'returns a new program' do
-      program = create(:program)
-      outcome = double(result: program, valid?: true)
-      attributes = {name: new_name}
-
-      # Why is this spec busted?
-      # params = {}.tap do |param|
-      #   param[:attributes] = attributes
-      #   param[:current_user] = current_user
-      #   param[:client] = panoptes_application_client
-      # end
-      # action_params = ActionController::Parameters.new(params)
-
-      # expect(Programs::Update).to receive(:run)
-      #   .with(action_params)
-      #   .and_return(outcome)
-
-      put :update, params: {id: program.id, data: {attributes: attributes}}, as: :json
```

Comment:
Did you mean to remove this entire test for PUT? I'm not too worried about going from mock-based tests to integration tests here, but it'd be nice to have at least 1 test per controller action that actually executes some code.

Reply:
Answer is twofold:
1. I did mean to remove it because I removed the actions for PUT and DELETE. There's no admin interface for programs yet and neither i2a or WC* will need them any time soon.

2. These are the specs (specifically, the ones checking if the op runs in the controller) were always extremely finicky for me. Things like rspec showing me two identical hashes but AI said there were "unexpected inputs". Basically endless iterations of something like this:
```       #<Programs::Create (class)> received :run with unexpected arguments
         expected: ({"client"=>#<Double Panoptes::Client>, "current_user"=>#<User id: 3, zooniverse_id: "9999", access_to...name", metadata: nil>, "attributes"=>{"slug"=>"wildcam-intro2Astro", "name"=>"Wildcam Intro2Astro"}})
              got: ({"attributes"=>{"name"=>"Wildcam Intro2Astro", "slug"=>"wildcam-intro2Astro"}, :client=>#<Double Pano...09-25 16:17:33", zooniverse_login: "login", zooniverse_display_name: "display_name", metadata: nil>})```
where I'm just stringifying keys or changing the order trying to get it to match and if felt extremely brittle. There must be a better, simpler way to confirm the op runs, but the controller tests are running them via request so testing the output seemed like enough.